### ISSUE TITLE
Provisioning headless server with config file

### ISSUE THREAD

**Issue by draoncc** [Anuken/Mindustry#111]
Automating running the server process isn't feasibly possible with the server requiring manual user input. While it is possible to interject stdout to react to prompts with e.g. (expect)[https://linux.die.net/man/1/expect], it would be cleaner and easier to provision a configuration file and daemonize the server with a flag.

For example, instead of running `java -jar server.jar`, and then typing in the difficulty and what map with which mode to host, you should be able to write `java -jar server.jar -d path/to/cfg`, where -d would daemonize the server and the first argument would be the config.

The config could simply be in JSON or a superset of JSON, with a structure such as the following:

```json
{
  "difficulty": "hard",
  "mode": "survival",
  "map": "fortress",
  ...
}
```

If the config is provided, ServerControl would not need to be initialized.

Another possibility could be to run the server with a sequential list of commands to be executed once it's spun up, by providing a string as the first argument like `java -jar server.jar -d -c "difficulty hard;host x"`.

**Comment by Anuken** [comment]
Implemented the latter option as of d396149521f6a488c02b8e5d63e78a8299fe3ea9, with the format `<command1> <command1-args>,<command2> <command2-args>`. As for daemonizing, I do not believe it is possible to do from within the application. You would have to use an external tool or script.

**Comment by halr9000** [comment]
@Anuken any way you could document this in the wiki and provide a few examples? Took me forever to find this issue. TIA
